numpy/conftest.py

```diff
@@ -114,6 +114,10 @@ def check_fpu_mode(request):
 
 @pytest.fixture(autouse=True)
 def add_np(doctest_namespace):
+    # NumPy should probably use more `.. plot::` directives even in
+    # examples, but as long as we don't, do not block on `plt.show()`
+    import matplotlib
+    matplotlib.use("Agg")
```

Comment:
@bsipocz where could this go and is this OK?  This fixes the `plt.show()` not popping up things.  There are some that can just be plot directives, but the histogram re-uses the data and has comments, etc. so I thought this is the easier path.

Maybe we need distinct runs, also so that we can make sure that `warnings` are not raised as errors?

Reply:
Indeed, this should be the fix for the mpl pop ups.

As for warnings, I think they really should be raised as errors, unless they are handled properly. For doctestplus we can use either `+IGNORE_WARNINGS` or `+SHOW_WARNINGS`